annotation/src/main/java/org/cbioportal/genome_nexus/annotation/domain/internal/EnsemblRepositoryImpl.java

```diff
@@ -0,0 +1,233 @@
+    private Map<String, List<EnsemblTranscript>> indexByGene;
+    private Map<String, List<EnsemblTranscript>> indexByProtein;
+    private Map<String, Map<String, List<EnsemblTranscript>>> indexByHugoSymbol;
+
+    @Autowired
+    public EnsemblRepositoryImpl(
+        @Value("${ensembl.biomart.transcripts:ensembl_biomart_transcripts.txt}") String biomartResourceURI,
+        @Value("${ensembl.biomart.hgnc_canonical_transcripts:ensembl_biomart_canonical_transcripts_per_hgnc.txt}") String canonicalTranscriptPerHugoSymbolResourceURI
+    )
+    {
+        this.biomartResourceURI = biomartResourceURI;
+        this.canonicalTranscriptPerHugoSymbolResourceURI = canonicalTranscriptPerHugoSymbolResourceURI;
+
+        // populate the list and build the indices
+        this.populateData();
+    }
+
+    public List<EnsemblTranscript> findAll()
+    {
+        return this.ensemblTranscriptList;
+    }
+
+    @Override
+    public List<EnsemblTranscript> findByTranscriptId(String transcriptId) {
+        return this.find(this.indexByTranscript, transcriptId);
+    }
+
+    @Override
+    public List<EnsemblTranscript> findByProteinId(String proteinId) {
+        return this.find(this.indexByProtein, proteinId);
+    }
+
+    @Override
+    public List<EnsemblTranscript> findByGeneId(String geneId) {
+        return this.find(this.indexByGene, geneId);
+    }
+
+    @Override
+    public List<EnsemblTranscript> findByHugoSymbol(String hugoSymbol, String isoformOverrideSource) {
+        Map<String, List<EnsemblTranscript>> canonicalTranscripts = indexByHugoSymbol.get(hugoSymbol);
+        if (canonicalTranscripts == null) {
+            return Collections.emptyList();
+        } else {
+            List<EnsemblTranscript> transcripts = canonicalTranscripts.get(isoformOverrideSource);
+            if (transcripts == null) {
+                return Collections.emptyList();
+            } else {
+                return transcripts;
+            }
+        }
+    }
+
+    private List<EnsemblTranscript> parseBiomartTSV()
+    {
+        BeanListProcessor<EnsemblTranscript> rowProcessor = new BeanListProcessor<>(EnsemblTranscript.class);
+
+        CsvParserSettings parserSettings = new CsvParserSettings();
+        parserSettings.setHeaderExtractionEnabled(true);
+        parserSettings.getFormat().setDelimiter('\t');
+        parserSettings.setRowProcessor(rowProcessor);
+
+        CsvParser parser = new CsvParser(parserSettings);
+        parser.parse(getReader(this.biomartResourceURI));
+
+        return rowProcessor.getBeans();
+    }
+
+
+    private void parseCanonicalTranscriptPerHugoSymbolTSV()
+    {
+        this.indexByHugoSymbol = new HashMap<>();
+        CsvParserSettings parserSettings = new CsvParserSettings();
+        parserSettings.setHeaderExtractionEnabled(true);
+        parserSettings.getFormat().setDelimiter('\t');
+        parserSettings.setRowProcessor(
+            new BeanProcessor<HugoSymbolRecord>(HugoSymbolRecord.class) {
```

Comment:
@onursumer isn't using the custom row processor better in terms of performance because you process them row by row, instead of first parsing the whole thing into memory and then going over the parsed list.